crmsh/ui_sbd.py

```diff
@@ -0,0 +1,524 @@
+        for node in self.cluster_nodes:
+            is_active = self.service_manager.service_is_active(constants.SBD_SERVICE, node)
+            is_active_str = "YES" if is_active else "NO"
+            is_enabled = self.service_manager.service_is_enabled(constants.SBD_SERVICE, node)
+            is_enabled_str = "YES" if is_enabled else "NO"
+            systemd_property = "ActiveEnterTimestamp" if is_active else "ActiveExitTimestamp"
+            since_str_prefix = "active since" if is_active else "disactive since"
+            systemctl_show_cmd = f"systemctl show {constants.SBD_SERVICE} --property={systemd_property} --value"
+            since = self.cluster_shell.get_stdout_or_raise_error(systemctl_show_cmd, node) or "N/A"
+            print(f"{node:<{max_node_len}}|{is_active_str:<{status_len}}|{is_enabled_str:<{status_len}}|{since_str_prefix}: {since}")
+        print()
+
+    def _print_watchdog_info(self):
+        padding = 2
+        max_node_len = max(len(node) for node in self.cluster_nodes) + padding
+
+        watchdog_sbd_re = r"\[[0-9]+\] (/dev/.*)\nIdentity: Busy: .*sbd.*\nDriver: (.*)"
+        device_list, driver_list, kernel_timeout_list = [], [], []
+        cluster_nodes = self.cluster_nodes[:]
+        for node in cluster_nodes[:]:
+            out = self.cluster_shell.get_stdout_or_raise_error("sbd query-watchdog", node)
+            res = re.search(watchdog_sbd_re, out)
+            if res:
+                device, driver = res.groups()
+                kernel_timeout = self.cluster_shell.get_stdout_or_raise_error("cat /proc/sys/kernel/watchdog_thresh", node)
+                device_list.append(device)
+                driver_list.append(driver)
+                kernel_timeout_list.append(kernel_timeout)
+            else:
+                logger.error("Failed to get watchdog info from %s", node)
+                cluster_nodes.remove(node)
+        if not cluster_nodes:
+            return
+
+        print("# Watchdog info:")
+        max_dev_len = max(len(dev) for dev in device_list) + padding
+        max_driver_len = max(len(driver) for driver in driver_list) + padding
+        print(f"{'Node':<{max_node_len}}|{'Device':<{max_dev_len}}|{'Driver':<{max_driver_len}}|Kernel Timeout")
+        for i, node in enumerate(cluster_nodes):
+            print(f"{node:<{max_node_len}}|{device_list[i]:<{max_dev_len}}|{driver_list[i]:<{max_driver_len}}|{kernel_timeout_list[i]}")
+        print()
+
+    def _print_sbd_agent_status(self):
+        if self.crm_mon_xml_parser.is_resource_configured(sbd.SBDManager.SBD_RA):
+            print("# Status of fence_sbd:")
+            sbd_id_list = self.crm_mon_xml_parser.get_resource_id_list_via_type(sbd.SBDManager.SBD_RA)
+            for sbd_id in sbd_id_list:
+                rc, output = self.cluster_shell.get_rc_and_error(None, None, f"crm resource status {sbd_id}", ignore_output=False)
```

Comment:
You need the output here. Then, I think the previous version, using a separated interface, is better